src/main/resources/common-services/CDAP/5.0.0/package/scripts/master.py

```diff
@@ -57,6 +60,62 @@ def start(self, env, upgrade_type=None):
+                                             if (output.startswith('[') and output.endswith(']')):
+                                                 if (output != '[]'):
+                                                     print(output)
+                                                     result = output[1:len(output)-1]
+                                                     # execute get /cdap/twill/<name>/instances/<last_output and save its output
+                                                     print(params.cdap_zookeeper_quorum)
+                                                     print(service)
+                                                     print(result)
+                                                     print('here executing the commands')
+                                                     print("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output")
+                                                     Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output", user='zookeeper')
+                                                     status, output = commands.getstatusoutput("tail -n1 /tmp/twill_" + service + "_instances_output")
+                                                     if status == 0:
+                                                         print(output)
+                                                         data = json.loads(output)
+                                                         container_id = data['data']['containerId']
+                                                         container_values = container_id.split('_')
+                                                         applicationId = "application_" + container_values[2] + "_" + container_values[3] 
+                                                         kinit_cmd = params.kinit_cmd_master
+                                                         kill_application_cmd = format("{kinit_cmd} yarn application -kill " + applicationId)
+                                                         print(kill_application_cmd)
+                                                         Execute(kill_application_cmd, user='cdap')
+                                                         print("applicationId : " + applicationId)
+                                                     Execute("rm -rf /tmp/twill_" + service + "_instances_output", user='zookeeper') 
+                                         Execute("rm -rf /tmp/twill_" + service + "_instances", user='zookeeper') 
+                    Execute("rm -rf /tmp/twill_master_services", user='zookeeper')                  
+        Execute("rm -rf /tmp/election_master_services", user='zookeeper')             
```

Comment:
1. remove the nested if statements.
2. remove one of the sleep command as it was not require and the other one remain as it is. I cant keep it inside loop as i dont know whether both the masters will be shutdown or not. Because in case of only one master shutdown from ambari this  should not stuck inside loop for keep waiting for the condition to get true. As my conition is a check on zookeeper node whether it become empty or not after cdap stop.
Worst case scenerio is if it didnt get empty even after my sleep period it will not stop the services whch is unlikely as my sleep command execute after the cdap stop complete and hence just wait for the zookeeper command to get completed once both cdap stops.
hence i am keeping that as it is